core/http/client/src/main/java/org/eclipse/rdf4j/http/client/SesameClientImpl.java

```diff
@@ -107,25 +125,26 @@ public synchronized SesameSession createSesameSession(String serverURL) {
 	 *-----------------*/
 
 	@Override
-	public synchronized void shutDown() {
-		if (executor != null) {
+	public void shutDown() {
+		try {
 			executor.shutdown();
-			executor = null;
 		}
-		if (dependentClient != null) {
-			HttpClientUtils.closeQuietly(dependentClient);
+		finally {
+			CloseableHttpClient toCloseDependentClient = dependentClient;
 			dependentClient = null;
+			if (toCloseDependentClient != null) {
+				HttpClientUtils.closeQuietly(toCloseDependentClient);
+			}
 		}
 	}
 
 	/**
-	 * (re)initializes the connection manager and HttpClient (if not already done), for example after a
-	 * shutdown has been invoked earlier. Invoking this method multiple times will have no effect.
+	 * No-op
+	 * 
+	 * @deprecated Create a new instance instead of trying to reactivate an old instance.
 	 */
-	public synchronized void initialize() {
-		if (executor == null) {
-			executor = Executors.newCachedThreadPool();
-		}
+	@Deprecated
+	public void initialize() {
```

Comment:
What's the consequence of this change for code that still _does_ rely on reinitialization? I didn't check but I assume you changed our own code to not do this (if it ever did)?